src/server/services/generation/generation.service.ts

```diff
@@ -337,11 +337,24 @@ const baseModelToOrchestration: Record<BaseModelSetType, string | undefined> = {
   SDXLDistilled: 'SDXL_Distilled',
 };
 
+async function checkGenerationAvailability(resources: CreateGenerationRequestInput['resources']) {
+  const data = await getResourceData(resources.map((x) => x.id));
+
+  return data.every((x) => !!x.generationCoverage?.covered);
+}
+
 export const createGenerationRequest = async ({
   userId,
   resources,
   params: { nsfw, negativePrompt, ...params },
 }: CreateGenerationRequestInput & { userId: number }) => {
+  if (!resources || resources.length === 0) throw throwBadRequestError('No resources provided');
```

Comment:
This condition is handled already on line 359

Reply:
Kinda, just doing an early throw to prevent running any other processes without resources